### ISSUE TITLE
Content item name "Constructor" breaks previewDeliveryClient query

### ISSUE THREAD

**Issue by boris-pocatko-fluent** [kontent-ai/delivery-sdk-js#388]
### Brief bug description

What went wrong?

![image](https://github.com/kontent-ai/delivery-sdk-js/assets/123047636/d87faf5a-94b3-4556-818b-90126c173fee)

The following query results in an error:

```
const query = await previewDeliveryClient
.items<ApprovedEmailDomainContentType>()
.queryConfig({ usePreviewMode: true })
.type(contentTypesCodename.approvedEmailDomain)
.elementsParameter(['approved_domain'])
.depthParameter(0)
.toAllPromise();
```

If I create a new content item with the content item name `Constructor`. 

### Repro steps

1. Create a content item name with the name `Constructor`
2. Run the above query

### Expected behavior

What is the correct behavior?

### Test environment

 - TypeScript

**Comment by IvanKiral** [comment]
Hey @boris-pocatko-fluent :)

right now, I don't see a problem with the name `Constructor`. Why do you think that this is the problem?

Also please could you provide me please with the look of the ApprovedEmailDomainContentType, so I could replicate the issue?

From the error I guess the problem relates to the LinkedItems. Do you have linked items in your type?

Thank you :)

**Comment by boris-pocatko-fluent** [comment]
Hey @IvanKiral ,

If I create any content item the query executes, if I create one with the name `Constructor` it fails. I assume as it may conflict with a reserved programming structure in the SDK. 

The content model is pretty basic without linked items: 
<img width="299" alt="image" src="https://github.com/kontent-ai/delivery-sdk-js/assets/123047636/96b0624e-2d6b-45eb-9fdb-fddd9c71029b">

**Comment by IvanKiral** [comment]
Thank you, 

I was able to reproduce the mistake, and you are right. SDK also cannot handle names of the object methods like `toString` and so on. We will look into it

**Comment by Enngage** [comment]
Hey @boris-pocatko-fluent , still using Kontent, are you? 😄 

I've released `14.7.0` which should fix this issue ;) Can you see if that helped in your case as well? Thanks for letting us know!
